app.js

```diff
@@ -27,12 +27,36 @@ if (getURLParameter('touch') === 'true') {
 // Check if HTML form is hardcoded or needs to be retrieved
 // note: when running this file in enketo-core-performance-monitor xform = 'null'
 if (xform && xform !== 'null') {
-    document.querySelector('.guidance').remove();
-    xform = /^https?:\/\//.test(xform) ? xform : `${location.origin}/${xform}`;
-    const transformerUrl = `http://${location.hostname}:8085/transform?xform=${xform}`;
-    fetch(transformerUrl)
-        .then((response) => response.json())
-        .then((survey) => {
+    (async () => {
+        const isRemote = /^https?:\/\//.test(xform);
+        const xformURL = isRemote ? xform : `${location.origin}/${xform}`;
+        const transformerUrl = `http://${location.hostname}:8085/transform?xform=${xformURL}`;
+
+        try {
+            document.querySelector('.guidance').remove();
```

Comment:
I can't quickly figure out what this does. Seems good that errors from it are now caught but also the error message about transform failing seems like it might be very unhelpful?

Reply:
Sorry, most of this is more Promise -> async/await noise. Most of this logic hasn't changed, including catching errors. I have also found the error message unhelpful which is why I interpolated the error into the alert. The other substantive change of course is requesting the cached form before requesting transformer.

For what it's worth, I suspect I'm one of, if not the, most frequent users of this. 🙂